submissions/maxim-zabolotny/friends-app/app.js

```diff
@@ -0,0 +1,215 @@
+            email : user.email,
+            location : user.location.city,
+            password : user.login.password,
+            gender : user.gender,
+            cell : user.cell
+        })
+    }
+}
+
+const createBtn = (page, type) => `<button class="btn-inline results__btn--${ type }" data-goto=${type === 'prev' ? page - 1 : page + 1}>
+                                        <span>Page ${ type === 'prev' ? page - 1 : page + 1 }</span>
+                                        <svg class="search__icon">
+                                            <use href="img/icons.svg#icon-triangle-${type === 'prev' ? 'left' : 'right'}"></use>
+                                        </svg>
+                                    </button>`;
+const renderBtn = (page, sumResults, resPerPage) => {
+    const pages = Math.ceil(sumResults/resPerPage);
+    let btn;
+    if(page === 1 && pages > 1){
+        btn = createBtn(page, 'next');
+    }else if (page === pages && pages > 1) {
+        btn = createBtn(page, 'prev');
+    }else if (page < pages){
+        btn = `${createBtn(page, 'next')} ${createBtn(page, 'prev')}`;
+    }else{
+        btn = null;
+    }
+    if( sumResults > 0 ){
+        if(DOM_NODES.btnWrapper) {
+            if(btn) DOM_NODES.btnWrapper.insertAdjacentHTML('beforeend', btn);
+        } else{
+            const btnWrapper = document.createElement('DIV');
+            btnWrapper.className = 'btn-wrapper';
+            if(btn) btnWrapper.insertAdjacentHTML('beforeend', btn);
+            DOM_NODES.target.appendChild(btnWrapper);
+        }
+    }
+};
+
+const renderCards = (cardsArray, currentPage = 1, resPerPage = 12) => {
+    const start = (currentPage - 1)*resPerPage;
+    const end = currentPage*resPerPage;
+    if(cardsArray && cardsArray.length !== 0){
+        const fragment = document.createDocumentFragment();
+        cardsArray.slice(start, end).forEach(item => fragment.appendChild(item.render()));
+        DOM_NODES.target.innerHTML = '';
+        DOM_NODES.target.appendChild(fragment);
+    }
+    renderBtn(currentPage, cardsArray.length, resPerPage)
+};
+
+const filter = (value) => {
+    renderCards(state.filteredUsers = state.allUsers.filter(({name}) => `${name.first} ${name.last}`.toLowerCase().includes(value.toLowerCase())))
+};
+
+const reset = () => {
+    state.sortedUsers = [...state.allUsers];
+    renderCards(state.filteredUsers = [...state.allUsers]);
+};
+const resetRadioButtons = () => {
+    reset();
+    Array.from(DOM_NODES.radioButtons).forEach(button => button.checked = false);
+    DOM_NODES.radioButtons[0].checked = true;
+};
+
+const sortingCards = (e) => {
+    const target = e.target.type;
+    if(!state.sortedUsers.length) state.sortedUsers = [...state.filteredUsers];
+    
+    if (target == 'radio') SORT[e.target.value]();
+};
+
+DOM_NODES.inputText.addEventListener("change", function (e) {
+    filter(e.target.value)
+});
+// DOM_NODES.inputText.addEventListener("focus", resetRadioButtons);
```

Comment:
Is this still needed? It's better to remove commented out code from final version